evalml/pipelines/components/estimators/regressors/exponential_smoothing_regressor.py

```diff
@@ -0,0 +1,141 @@
+"""Holt-Winters Exponential Smoothing Forecaster."""
+import numpy as np
+import pandas as pd
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import import_or_raise, infer_feature_types
+
+
+class ExponentialSmoothingRegressor(Estimator):
+    """Holt-Winters Exponential Smoothing Forecaster.
+
+    Currently ExponentialSmoothingRegressor isn't supported via conda install. It's recommended that it be installed via PyPI.
+
+    Args:
+        trend (str): Type of trend component. Defaults to None.
+        damped_trend (bool): If the trend component should be damped. Defaults to False.
+        seasonal (str): Type of seasonal component. Takes one of {“add”, “mul”, “additive”, “multiplicative”, None}. Defaults to None.
+        sp (int): The number of seasonal periods to consider. Defaults to 1.
+        use_boxcox (bool, str): If the Box-Cox transform should be applied to the data first. If ‘log’ then apply the log. Defaults to False.
+        n_jobs (int or None): Non-negative integer describing level of parallelism used for pipelines. Defaults to -1.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Exponential Smoothing Regressor"
+    hyperparameter_ranges = {
+        "trend": [None, "additive", "multiplicative"],
+        "damped_trend": [True, False],
+        "seasonal": [None, "additive", "multiplicative"],
+        "sp": Integer(2, 8),
+    }
+    """{
+        "trend": [None, "additive", "multiplicative"],
+        "damped_trend": [True, False],
+        "seasonal": ["additive", "multiplicative"],
+        "sp": Integer(1, 12),
+    }"""
+    model_family = ModelFamily.EXPONENTIAL_SMOOTHING
+    """ModelFamily.EXPONENTIAL_SMOOTHING"""
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+    """[ProblemTypes.TIME_SERIES_REGRESSION]"""
+
+    def __init__(
+        self,
+        trend=None,
+        damped_trend=False,
+        seasonal=None,
+        sp=2,
```

Comment:
Given that the default value of seasonal is None, shouldn't the default value of sp be None too? Is sp used only if seasonal is not None?

Reply:
I would have thought so as well, but there's something in the sktime implementation that enforces otherwise. I can't tell exactly where sp is used when seasonal is None, but it fails otherwise.